lucene/core/src/java/org/apache/lucene/search/TopFieldCollector.java

```diff
@@ -69,6 +69,107 @@ public void setScorer(Scorable scorer) throws IOException {
     }
   }
 
+  private abstract class TopFieldLeafCollector extends MultiComparatorLeafCollector {
```

Comment:
This source (`TopFieldCollector.java`) has really become too crazy ... we should at some point pull all these inner classes out to their own source files ... thank you for opening https://issues.apache.org/jira/browse/LUCENE-9202

Reply:
hmm maybe we can do more there, but given how this is stacked on the refactoring PR in that issue I'd rather avoid to much more refactoring rn.